### ISSUE TITLE
Failed migration updating to Craft 4

### ISSUE THREAD

**Issue by darylknight** [engram-design/CPNav#103]
**Description**
I've just tried updating a fully up to date Craft 3 site to Craft 4, and got this error message while running migrations during the update to Craft 4.

Sorry I'm not sure what other information to give you - let me know?

```
Backing up the database ... done
*** applying m220409_000000_craft_4
    > update in {{%cpnav_navigation}} ... done (time: 0.000s)
    > dropping {{%cpnav_pending_navigations}} if it exists ... done (time: 0.157s)
Exception: Calling unknown method: craft\console\Request::getSegment() (/sitename/vendor/yiisoft/yii2/base/Component.php:300)
#0 /sitename/vendor/nystudio107/craft-seomatic/src/Seomatic.php(379): yii\base\Component->__call('getSegment', Array)
#1 /sitename/vendor/verbb/cp-nav/src/migrations/m220409_000000_craft_4.php(68): nystudio107\seomatic\Seomatic->getCpNavItem()
#2 /sitename/vendor/craftcms/cms/src/db/Migration.php(49): verbb\cpnav\migrations\m220409_000000_craft_4->safeUp()
#3 /sitename/vendor/yiisoft/yii2/console/controllers/BaseMigrateController.php(758): craft\db\Migration->up()
#4 /sitename/vendor/craftcms/cms/src/console/controllers/MigrateController.php(360): yii\console\controllers\BaseMigrateController->migrateUp('m220409_000000_...')
#5 [internal function]: craft\console\controllers\MigrateController->actionAll()
#6 /sitename/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#7 /sitename/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#8 /sitename/vendor/yiisoft/yii2/console/Controller.php(182): yii\base\Controller->runAction('all', Array)
#9 /sitename/vendor/yiisoft/yii2/base/Module.php(552): yii\console\Controller->runAction('all', Array)
#10 /sitename/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('migrate/all', Array)
#11 /sitename/vendor/craftcms/cms/src/console/Application.php(90): yii\console\Application->runAction('migrate/all', Array)
#12 /sitename/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('migrate/all', Array)
#13 /sitename/vendor/craftcms/cms/src/console/Application.php(121): yii\console\Application->handleRequest(Object(craft\console\Request))
#14 /sitename/vendor/yiisoft/yii2/base/Application.php(384): craft\console\Application->handleRequest(Object(craft\console\Request))
#15 /sitename/craft(13): yii\base\Application->run()
#16 {main}
*** failed to apply m220409_000000_craft_4 (time: 0.264s)

0 from 10 migrations were applied.

Migration failed. The rest of the migrations are canceled.
```

**Additional info**
- Plugin version: 2.7.2 > 4.0.1
- Craft version: 3.7.44 > ^4.0.0
- Multi-site: No

**Comment by engram-design** [comment]
Seems like this is a conflict with SEOmatic. Looking into it.

**Comment by engram-design** [comment]
Should be fixed in [4.0.2](https://github.com/verbb/cp-nav/releases/tag/4.0.2)

**Comment by darylknight** [comment]
Yep that works - thanks!
